server/graphql/v2/mutation/TransactionMutations.ts

```diff
@@ -23,6 +26,56 @@ const transactionMutations = {
       return legacyRefundTransaction(undefined, { id: transaction.id }, req);
     },
   },
+  rejectTransaction: {
+    type: new GraphQLNonNull(Transaction),
+    description: 'Rejects transaction, removes member from Collective, and sends a message to the contributor',
+    args: {
+      transaction: {
+        type: new GraphQLNonNull(TransactionReferenceInput),
+        description: 'Reference of the transaction to refund',
+      },
+    },
+    async resolve(_, args, req): Promise<typeof Transaction> {
+      if (!req.remoteUser) {
+        throw new Unauthorized();
+      }
+      // get transaction info
+      const transaction = await fetchTransactionWithReference(args.transaction);
+      // refund transaction and set status
+      const refundedTransaction = await legacyRefundTransaction(undefined, { id: transaction.id }, req);
+      const orderToUpdate = await models.Order.update(
+        {
+          status: orderStatus.REJECTED,
+        },
+        {
+          where: { id: refundedTransaction.OrderId },
+          returning: true,
+        },
+      );
+      if (!orderToUpdate) {
+        throw new NotFound('Order not found');
+      }
+      // get membership info
+      const toAccount = await fetchAccountWithReference({ legacyId: transaction.CollectiveId });
+      const fromAccount = await fetchAccountWithReference({ legacyId: transaction.FromCollectiveId });
+      const membership = await models.Member.findOne({
```

Comment:
Someone can have multiple memberships to the same collective if they donate through multiple tiers. So either:
- We want to delete them all, then we need to use a `findAll`
- We want to delete just the one matching this contribution, then we need to add `TierId: order.TierId` to the query

Reply:
Interesting. I think we'd want to delete them all right? Because the point is that the person rejecting *doesn't* want this backer in their Collective at all. 

Would that be an issue to delete the memberships associated with orders/transactions that haven't also been rejected?

Reply:
> I think we'd want to delete them all right

Seems like a good call. Then we shouldn't throw if the membership doesn't exist.

Also, may be worth using `models.Member.delete({ where ... })` to make only one SQL query, and to invalidate the cache only one time (must be done manually though `invalidateContributorsCache`, as models `hooks` are not triggered when using the `Model.Member.update` syntax.

Reply:
I've just seen this comment, I will update the code after standup

Reply:
Sorry, have looked at the Sequelize docs and tried `models.Member.delete` and I get an error it isn't a function and it doesn't say this exists in the docs. It says to use `destroy()`.